selftests/unit/test_data_structures.py

```diff
@@ -29,6 +29,16 @@ def test_geometric_mean(self):
         self.assertEqual(data_structures.geometric_mean(xrange(1, 180)),
                          67.1555819421869)
 
+    def test_compare_matrices(self):
+        """
+        Verify the correct value is produced when comparing matrices.
+        """
+        matrix1 = [[4, 51.7, 60], [6, 0, 0]]
```

Comment:
Although this is a valid matrix and a nice test, I'd add yet another check for header (with explanation to avoid confusion):

```
# Note that first row contains header in first column, while the
# second contains only values (for testing purposes)
matrix1 = [["header", 51.7, 60], [1, 0, 0]]
matrix2 = [["header", 57.2, 54], [2, 51, 0]]
```